### PR TITLE
feat: update tool versions

### PR DESCRIPTION
**Pull request by kaldorn**
## What this PR does / why we need it
Bumps tooling versions.

## Jira ID

[DT-3447]

## Notes for your reviewers

[DT-3447]: https://outreach-io.atlassian.net/browse/DT-3447?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: versions.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 grpcui: 1.3.1
-jsonnetfmt: 4906958414ed9617e3fe5acac5752f6f8426551c # use a tagged version whenever they release past v0.18.0 (https://github.com/google/go-jsonnet)
-goimports: 0.1.12
-delve: 1.9.1
-gotestsum: 1.7.0
-lintroller: 1.15.1
+jsonnetfmt: v1.19.1
```
Comment (jaredallard) [inline versions.yaml:2]: No `v` pls

Comment (jaredallard) [reply versions.yaml:2]: (might need to go fix jsonnetfmt code to support no `v`, but the convention is no `v` here)

File: .tool-versions [hunk 2]
```diff
@@ -16,7 +16,7 @@ protoc 21.5
 ## <<Stencil::Block(toolver)>>
 # The below tools are used by all repositories when using devbase
 mage 1.14.0
-golangci-lint 1.49.0
-shellcheck 0.8.0
-shfmt 3.4.2
+golangci-lint 1.51.1
```
[the remark this answers is not in the document]
Comment (kaldorn) [reply .tool-versions:19]: Yes, what's the process for enabling those?

I saw this line and figured that we have a different process for enabling new linters: https://github.com/getoutreach/devbase/blob/f1ed94615ba2c8334b6c8b69ba5010abacd1f09f/scripts/golangci.yml#L47

File: versions.yaml (hunk 1 above)
Comment (kaldorn) [reply versions.yaml]: Seems like it works without the v.

File: .tool-versions (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaredallard) [reply .tool-versions:19]: Oh I didn't realize we had fixed this, we're good then :D

**Review by jaredallard — APPROVED** [review]

**Pull request merged**